packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -845,6 +846,55 @@ export class DetailsListBase extends React.Component<IDetailsListProps, IDetails
     return 0;
   }
 
+  public updateColumn(column: IColumn, newProps: { width?: number; newColumnIndex?: number }) {
+    const NO_COLUMNS: IColumn[] = [];
+
+    const { columns = NO_COLUMNS, selectionMode, checkboxVisibility, columnReorderOptions } = this.props;
+    const { width, newColumnIndex } = newProps;
+    const index = columns.findIndex(col => col.key === column.key);
+
+    if (width) {
+      this._onColumnResized(column, width, index!);
+    }
+
+    if (newColumnIndex) {
```

Comment:
Actually if I'm following the logic right, a further simplification would be to exit at this point if `newColumnIndex === undefined || !columnReorderOptions`. (the === undefined is because 0 is falsey but I'm guessing could be a valid value)